distributed/client.py

```diff
@@ -3961,6 +3961,8 @@ def register_worker_callbacks(self, setup=None):
         return self.register_worker_plugin(_WorkerSetupPlugin(setup))
 
     async def _register_worker_plugin(self, plugin=None, name=None):
+        if getattr(plugin, "name", None) is None:
+            plugin.name = name
```

Comment:
Hrm, this seems a little janky.  We could also investigate changing `Scheduler.worker_plugins` from a list to a dict, or storing the plugins themselves as dictionaries like `{"plugin": my_plugin, "name": name}`.  Those probably both have problems (we lose ordering in the first case, and it's not clear what to do when we don't have a name).

Reply:
Hrm, maybe not though.  I guess if it gets the job done?

Reply:
With the conditional check we don't need the additional `is None`

Reply:
@mrocklin - definitely janky :) but I didn't want to change existing behavior if the plugin has a name, so I think if we want to pass a dict, then we should use `plugin.name` as the default when name isn't provided as a kwarg. Which feels pretty similar to me. 

@quasiben - You don't think the name could be something that evaluates to False (like 0)?

Reply:
Oh! I wasn't thinking about numerical names.  Thanks for the correction

Reply:
So is this an ok solution? Should I go ahead and add/fix tests?

Reply:
This seems OK to me. Changing the container from a list to dict also seems sensible, but probably will be better as a followup.